step-templates/put-server-back-in-azure-load-balancer.json

```diff
@@ -0,0 +1,68 @@
+{
+  "Id": "540de6f6-f1c6-4e1c-9062-9e78570074f6",
+  "Name": "Put Server Back In Azure Load Balancer",
+  "Description": "Put Server Back In Azure Load Balancer",
+  "ActionType": "Octopus.AzurePowerShell",
+  "Version": 1,
+  "CommunityActionTemplateId": null,
+  "Properties": {
+    "Octopus.Action.Script.ScriptSource": "Inline",
+    "Octopus.Action.Script.ScriptBody": "#region Verify variables\n\n#Verify psbilbAzureSubscription is not null.\nIf ([string]::IsNullOrEmpty($OctopusParameters['psbilbAzureSubscription']))\n{\n  Throw 'Azure Subscription cannot be null.'\n}\n$azureSubscription = $OctopusParameters['psbilbAzureSubscription']\nWrite-Host ('Azure Subscription: ' + $azureSubscription)\n\n#Verify psbilbAzureResourceGroup is not null.\nIf ([string]::IsNullOrEmpty($OctopusParameters['psbilbAzureResourceGroup']))\n{\n  Throw 'Azure Resource Group cannot be null.'\n}\n$azureResourceGroup = $OctopusParameters['psbilbAzureResourceGroup']\nWrite-Host ('Azure Resource Group: ' + $azureResourceGroup)\n\n\n#Verify psbilbAzureMachineName is not null.\nIf ([string]::IsNullOrEmpty($OctopusParameters['psbilbAzureMachineName']))\n{\n  Throw 'Azure Machine Name cannot be null.'\n}\n$azureMachineName = $OctopusParameters['psbilbAzureMachineName']\nWrite-Host ('Azure Machine Name: ' + $azureMachineName)\n\n#Verify psbilbAzureLoadBalancer is not null.\nIf ([string]::IsNullOrEmpty($OctopusParameters['psbilbAzureLoadBalancer']))\n{\n  Throw 'Azure Load Balancer cannot be null.'\n}\n$azureLoadBalancer = $OctopusParameters['psbilbAzureLoadBalancer']\nWrite-Host ('Azure Load Balancer: ' + $azureLoadBalancer)\n\n#endregion\n\n\n#region Process\n\nTry\n{\n  Get-AzureRmContext | Out-Null\n}\nCatch\n{\n  If ($_.Exception.ToString() -like \"*Run Login-AzureRmAccount to login.*\")\n  {\n    Login-AzureRmAccount\n  }\n}\n\nSet-AzureRmContext -SubscriptionName $azureSubscription\n\n$azureVM = Get-AzureRmVM -ResourceGroupName $azureResourceGroup -Name $azureMachineName\nIf (!$azureVM)\n{\n  Throw 'Could not retrieve server from Azure needed to remove from Load Balancer.'\n}\n\n$nic = (Get-AzureRmNetworkInterface -ResourceGroupName $azureResourceGroup | Where-Object {$_.VirtualMachine.Id -eq $azureVM.Id})\nIf (!$nic)\n{\n  Throw 'Could not retrieve NIC from Azure needed to remove from Load Balancer.'\n}\n\n$loadBalancer = Get-AzureRmLoadBalancer -Name $azureLoadBalancer -ResourceGroupName $azureResourceGroup\nIf (!$loadBalancer)\n{\n  Throw 'Could not retrieve Load Balancer info from Azure.'\n}\n\n$nic.IpConfigurations[0].LoadBalancerBackendAddressPools = $loadBalancer.BackendAddressPools\nSet-AzureRmNetworkInterface -NetworkInterface $nic\n\n#endregion\n",
```

Comment:
```
Try
{
  Get-AzureRmContext | Out-Null
}
Catch
{
  If ($_.Exception.ToString() -like "*Run Login-AzureRmAccount to login.*")
  {
    Login-AzureRmAccount
  }
}
```
I don't think this should happen as our bootstrap script handles all the authentication to Azure, **but** could there be a chance here that the `Login-AzureRmAccount` would open a dialog and cause the deployment to hang?